src/com/dotcms/plugin/saml/v3/rest/api/v1/IdpConfigHelper.java

```diff
@@ -0,0 +1,185 @@
+            //Delete files from FS.
+            deleteFile(idpConfig.getPrivateKey());
+            deleteFile(idpConfig.getPublicCert());
+            deleteFile(idpConfig.getIdPMetadataFile());
+
+        } else {
+            Logger.warn(this, "IdpConfig with Id: " + idpConfig.getId() + "no longer exists in file.");
+        }
+    } // deleteIdpConfig.
+
+    public void setDefaultIdpConfig(String idpConfigId) throws IOException, JSONException, DotDataException {
+        List<IdpConfig> idpConfigList = getIdpConfigs();
+
+        final IdpConfig idpConfig = new IdpConfig();
+        idpConfig.setId(idpConfigId);
+
+        if (idpConfigList.contains(idpConfig)){
+            IdpConfigWriterReader.writeDefaultIdpConfigId(idpConfigId, idpfilePath);
+        }
+        else {
+            Logger.error(this, "IdpConfig with Id: " + idpConfig.getId() + "no longer exists in file.");
+            throw new DotDataException("IdpConfig with Id: " + idpConfig.getId() + "no longer exists in file.");
+        }
+    } // setDefaultIdpConfig.
+
+    public String getDefaultIdpConfigId() throws IOException, JSONException {
+        return IdpConfigWriterReader.readDefaultIdpConfigId(new File(idpfilePath));
+    } // setDefaultIdpConfig.
+
+    public void saveDisabledSiteIds(Map<String, String> disablebSitesMap) throws IOException, JSONException {
+        IdpConfigWriterReader.writeDisabledSIteIds(disablebSitesMap, idpfilePath);
+    } // saveDisabledSiteIds.
+
+    public Map<String, String> getDisabledSiteIds() throws IOException, JSONException {
+        return IdpConfigWriterReader.readDisabledSiteIds(new File(idpfilePath));
+    } // getDisabledSiteIds.
+
+    private IdpConfig renameIdpConfigFiles(IdpConfig idpConfig) throws IOException {
+        if (UtilMethods.isSet(idpConfig.getPrivateKey())){
+            idpConfig.setPrivateKey(writeCertFile(idpConfig.getPrivateKey(), idpConfig.getId() + ".key"));
+        }
+        if (UtilMethods.isSet(idpConfig.getPublicCert())){
+            idpConfig.setPublicCert(writeCertFile(idpConfig.getPublicCert(), idpConfig.getId() + ".crt"));
+        }
+        if (UtilMethods.isSet(idpConfig.getIdPMetadataFile())){
+            idpConfig.setIdPMetadataFile(writeMetadataFile(idpConfig.getIdPMetadataFile(), idpConfig.getId() + ".xml"));
+        }
+
+        return idpConfig;
+    } // renameIdpConfigFiles.
+
+    private File writeCertFile(File sourceFile, String fileName) throws IOException{
+        return this.writeFile(sourceFile, this.certsParentPath, fileName);
+    } // writeCertFile.
+
+    private File writeMetadataFile(File sourceFile, String fileName) throws IOException{
+        return this.writeFile(sourceFile, this.metadataParentPath, fileName);
+    } // writeMetadataFile.
+
+    private File writeFile(File sourceFile, String parentPath, String fileName) throws IOException{
+        File targetFile = new File(parentPath + fileName);
+
+        if (!targetFile.exists()) {
```

Comment:
and can write

Reply:
Well, in this case I am just checking if exists in order to create it.